overlord/configstate/configcore/services.go

```diff
@@ -372,8 +372,20 @@ func handleServiceConfigSSHListen(dev sysconfig.Device, tr ConfGetter, opts *fsO
 
 	if opts == nil {
 		sysd := systemd.New(systemd.SystemMode, &sysdLogger{})
-		if err := sysd.ReloadOrRestart([]string{"ssh.service"}); err != nil {
-			return err
+		// From 22.10 sshd now uses socket based activation. This changes how to reload
+		// ssh configuration
+		// Discussion here: https://discourse.ubuntu.com/t/sshd-now-uses-socket-based-activation-ubuntu-22-10-and-later/30189/9
+		if enabled, err := sysd.IsEnabled("ssh.socket"); err == nil && enabled {
+			if err := sysd.DaemonReload(); err != nil {
+				return err
+			}
+			if err := sysd.Restart([]string{"ssh.socket"}); err != nil {
+				return err
+			}
+		} else {
+			if err := sysd.ReloadOrRestart([]string{"ssh.service"}); err != nil {
```

Comment:
Should this not be done always regardless of requiring daemon reload, to ensure we sync service config?

Reply:
Nope, the other approach is documented exactly to do it like that - and it works by restarting the socket. The reason it works is because of `daemon-reload`, because the socket is actually generated by a systemd generator

Reply:
This is from the documentation:
```
# When systemd socket activation is used (the default), the socket
# configuration must be re-generated after changing Port, AddressFamily, or
# ListenAddress.
#
# For changes to take effect, run:
#
# systemctl daemon-reload
# systemctl restart ssh.socket
```

Reply:
![image](https://github.com/user-attachments/assets/4a13694f-e0f0-493c-9d99-01863ff6e5d9)
![image](https://github.com/user-attachments/assets/cbb6d6c7-788a-463a-97ee-6c50a51ad142)

Reply:
Yea I was wrong in regards to that, but following the documentation provided for making changes take effect works, and let us have that as-is.